packages/arb-token-bridge-ui/src/token-bridge-sdk/hooks/useArbTokenBridge.ts

```diff
@@ -54,6 +62,17 @@ export function getExecutedMessagesCacheKey({
     : `l2ChainId: ${l2ChainId}, position: ${event.position.toString()}`
 }
 
+export function getUniqueIdOrHashFromEvent(
+  event: L2ToL1EventResult
+): BigNumber {
+  // Nitro
+  if ('hash' in event) {
```

Comment:
Just to make sure: if `hash` property exists but is `undefined` this will return true. Do we want that?

Reply:
we can import `import { L2ToL1TxEvent as NitroL2ToL1TransactionEvent } from '@arbitrum/sdk/dist/lib/abi/ArbSys'` and cast to that

Reply:
```suggestion
  const nitroEvent = event as EventArgs<NitroL2ToL1TransactionEvent>
  const classicEvent = event as EventArgs<ClassicL2ToL1TransactionEvent>

  // Nitro
  if (typeof nitroEvent.hash !== 'undefined') {
    return nitroEvent.hash
  }
  // Classic
  return classicEvent.uniqueId
```

Reply:
Update: reverted to original implementation after final review.